### PR TITLE
Upgrade github.com/hashicorp/go-getter to v2 for remove indirect github.com/aws/aws-sdk-go reference

### PR DESCRIPTION
**Pull request by bozaro**
## Description

Related to #1432.

```
go mod why -m github.com/aws/aws-sdk-go
# github.com/aws/aws-sdk-go
github.com/gruntwork-io/terratest/modules/opa
github.com/hashicorp/go-getter
github.com/aws/aws-sdk-go/aws
```

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Upgrade github.com/hashicorp/go-getter to v2.

### CODE REVIEW

File: modules/opa/download_policy.go [hunk 1]
```diff
@@ -66,9 +62,32 @@ func DownloadPolicyE(t testing.TestingT, rulePath string) (string, error) {
...
-	if err := getter.GetAny(tempDir, baseDir); err != nil {
+	if _, err := getter.GetAny(context.Background(), tempDir, baseDir); err != nil {
 		return "", err
 	}
 	policyDirCache.Store(baseDir, tempDir)
 	return filepath.Join(tempDir, subDir), nil
 }
+
+// detectGetter find suitable getter.Getter by rulePath.
+func detectGetter(rulePath string) (getter.Getter, error) {
```
Comment (james03160927) [inline modules/opa/download_policy.go:73]: Would you please create unit test for this change to make sure we can continue to validate whether this code works going forward and now?

File: modules/opa/download_policy.go [hunk 2]
```diff
@@ -66,9 +62,32 @@ func DownloadPolicyE(t testing.TestingT, rulePath string) (string, error) {
...
 	logger.Default.Logf(t, "Downloading %s to temp dir %s", rulePath, tempDir)
-	if err := getter.GetAny(tempDir, baseDir); err != nil {
+	if _, err := getter.GetAny(context.Background(), tempDir, baseDir); err != nil {
 		return "", err
 	}
 	policyDirCache.Store(baseDir, tempDir)
 	return filepath.Join(tempDir, subDir), nil
 }
+
+// detectGetter find suitable getter.Getter by rulePath.
```
Comment (james03160927) [inline modules/opa/download_policy.go:72]: Can you elaborate what this mean? What does it mean to find suitable `getter.Getter`?

File: modules/opa/download_policy.go (hunk 1 above)
Comment (bozaro) [reply modules/opa/download_policy.go]: I thought about it a bit and rewrote the patch. This variant should be more straightforward.

**Review by james03160927 — APPROVED** [review]
The overall change LGTM. The failing tests do not seem to be relevant to this change.

**Review by denis256 — APPROVED** [review]

**Pull request merged**